src/libsyntax/parse/parser.rs

```diff
@@ -196,14 +197,12 @@ enum PrevTokenKind {
 #[derive(Clone)]
 pub struct Parser<'a> {
     pub sess: &'a ParseSess,
-    /// the current token:
-    pub token: token::TokenKind,
-    /// the span of the current token:
-    pub span: Span,
-    /// the span of the previous token:
+    /// The current token.
+    pub token: Token,
+    /// The span of the previous token.
     meta_var_span: Option<Span>,
     pub prev_span: Span,
-    /// the previous token kind
+    /// The previous token kind.
```

Comment:
merge with `prev_span`

Reply:
`PrevTokenKind` is not `TokenKind`, it's a different (simplified) thing.
We could probably keep the full previous token, but that's not strictly necessary apparently.

Also, `prev_span` is not really the previous `token.span`, it's a bit messy due to macros (see `meta_var_span`) and needs some investigation and cleanup.

Reply:
Makes sense, leave a FIXME on `prev_span` then?